### ISSUE TITLE
If ReduceLROnPlateau lr scheduler is passed inside a dict it won't work

### ISSUE THREAD

**Issue by jpvcaetano** [PyTorchLightning/pytorch-lightning#1215]
## 🐛 Bug

If a ReduceLROnPlateau lr scheduler is passed inside a dict the atribute scheduler['reduce_on_plateau'] will not be set to true.

### To Reproduce

ReduceLROnPlateau lr schedule inside a dictionary and try to train. The error will be something about that there is no metric in step().

#### Simple fix

In line 712 of trainer.py replace
`scheduler['reduce_on_plateau'] = \
                    isinstance(scheduler, optim.lr_scheduler.ReduceLROnPlateau)`
by
`scheduler['reduce_on_plateau'] = \
                    isinstance(scheduler['scheduler'], optim.lr_scheduler.ReduceLROnPlateau)`

**Comment by SkafteNicki** [comment]
this is fixed on master:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3be81cb54ebf2b5425cae09327e852bea0e7c492/pytorch_lightning/trainer/trainer.py#L723-L730
